gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/SweNetworkShifter.java

```diff
@@ -149,13 +152,24 @@ public void shiftNetwork(double stepValue, Network network) throws GlskLimitatio
 
             // Step 5: Reset current variant with initial state
             network.getVariantManager().setWorkingVariant(initialVariantId);
+            network.getVariantManager().removeVariant(processedVariantId);
             network.getVariantManager().removeVariant(workingVariantCopyId);
         } finally {
+            // revert connections of TWT on generators that were not used by the scaling
+            scalableGeneratorConnector.revertUnnecessaryChanges(network);
```

Comment:
Does setting the variant to the initialVariant not already reset terminal connections and targetP?

Reply:
we want to revert only the ones that were not connected by scaling and set to a specific targetP